### PR TITLE
Skip s3 tests if no region is specified

### LINKED ISSUE #144: S3 storagedriver URL problem in test suite
There appears to be a problem with the generated s3 URLs. Am I missing something?

```
$ export S3_ENCRYPT=false
$ export S3_BUCKET=mydockerregistry
$ export AWS_ACCESS_SECRET=XXXXXXXXXXX
$ export AWS_ACCESS_KEY=YYYYYYYYYYYYYY
$ go test -test.timeout=99h -test.short=true -check.vv -check.f TestContinueStreamAppendLarge
START: /Users/alp/workspace/gopath-docker-distribution/src/github.com/docker/distribution/storagedriver/testsuites/testsuites.go:95: DriverSuite.SetUpSuite
/Users/alp/workspace/gopath-docker-distribution/src/github.com/docker/distribution/storagedriver/testsuites/testsuites.go:100:
    c.Assert(err, check.IsNil)
... value *url.Error = &url.Error{Op:"Get", URL:"///mydockerregistry/?delimiter=&marker=&max-keys=1&prefix=%2Fvar%2Ffolders%2Fl0%2Fkj9cb1t95f5f73d3mbwhxdgm0000gn%2FT%2Fdriver-320629941", Err:(*http.badStringError)(0xc20801eae0)} ("Get ///mydockerregistry/?delimiter=&marker=&max-keys=1&prefix=%2Fvar%2Ffolders%2Fl0%2Fkj9cb1t95f5f73d3mbwhxdgm0000gn%2FT%2Fdriver-320629941: unsupported protocol scheme \"\"")

FAIL: /Users/alp/workspace/gopath-docker-distribution/src/github.com/docker/distribution/storagedriver/testsuites/testsuites.go:95: DriverSuite.SetUpSuite

START: /Users/alp/workspace/gopath-docker-distribution/src/github.com/docker/distribution/storagedriver/testsuites/testsuites.go:380: DriverSuite.TestContinueStreamAppendLarge
MISS: /Users/alp/workspace/gopath-docker-distribution/src/github.com/docker/distribution/storagedriver/testsuites/testsuites.go:380: DriverSuite.TestContinueStreamAppendLarge

START: /Users/alp/workspace/gopath-docker-distribution/src/github.com/docker/distribution/storagedriver/testsuites/testsuites.go:105: DriverSuite.TearDownSuite
PASS: /Users/alp/workspace/gopath-docker-distribution/src/github.com/docker/distribution/storagedriver/testsuites/testsuites.go:105: DriverSuite.TearDownSuite	0.000s

OOPS: 0 passed, 1 FAILED, 1 MISSED
--- FAIL: Test (0.01s)
FAIL
exit status 1
FAIL	github.com/docker/distribution/storagedriver/s3	0.027s
```

#question @stevvooe @AndreyKostov

### PR DESCRIPTION
**Pull request by AndreyKostov**
Closes #144.

### CODE REVIEW

File: storagedriver/s3/s3_test.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func init() {
 
 	// Skip S3 storage driver tests if environment variable parameters are not provided
 	skipCheck := func() string {
-		if accessKey == "" || secretKey == "" || bucket == "" || encrypt == "" {
+		if accessKey == "" || secretKey == "" || region == "" || bucket == "" || encrypt == "" {
 			return "Must set AWS_ACCESS_KEY, AWS_SECRET_KEY, S3_BUCKET, and S3_ENCRYPT to run S3 tests"
```
Comment (ahmetalpbalkan) [inline storagedriver/s3/s3_test.go]: you gotta add S3_REGION here too

Comment (AndreyKostov) [inline storagedriver/s3/s3_test.go]: Right... that's why you don't do things hastily :)

**Comment by ahmetalpbalkan** [comment]
lgtm

**Comment by stevvooe** [comment]
LGTM!

**Pull request merged**